### ISSUE TITLE
Lyx Lyz Server error - Help please!!

### ISSUE THREAD

**Issue by richwain** [willsALMANJ/lyz#32]
Hi
First time user of GitHub - any advice greatly received.

I've been using Lyx and Lyz with Zotero for years without a problem, however, I am now trying to make some amendments to my Thesis by adding new references and am getting repeated errors regarding connecting to the server and lyxpipe compatibility.

With my LyX document open, I try to insert a citation using the 'cite in lyx' option from LyZ. I then get the following:

SERVER ERROR:
[Exception... "Component returned failure code: 0x80570018 (NS_ERROR_XPC_BAD_IID) [nsIJSCID.createInstance]"  nsresult: "0x80570018 (NS_ERROR_XPC_BAD_IID)"  location: "JS frame :: chrome://lyz/content/lyz.js :: lyxPipeInit :: line 148"  data: no]

then:

Could not contact server at: ~/.lyxpipe

then:

Could not retrieve document name.

I have double-checked my settings in LyX - LyXServer pipe is: ~/.lyxpipe
And also my Lyx Server in LyZ is: ~/.lyxpipe

Tearing my hair out and don't know where to go next with this. Any advice much appreciated. BTW i'm a novice with code, so please be as clear as possible and assume I know nothing!!!

Best Regards

Richard

**Comment by richwain** [comment]
BTW my versions are:
LyX 2.3.0
Zotero 5.0.85
LyZ 3.0.1

**Comment by willsALMANJ** [comment]
Hmmm, are you using macOS?

Could you give me any more information about how this issue started? Had your set up worked recently for you or had you not tried to use LyZ recently and hit this problem as soon as you tried to start using it again?

**Comment by richwain** [comment]
Hi

Yes, using Mac OS - Mojave

It had been working fine up until around 6-8 months ago. That was the last time I used it - had a break from research!

I’ve managed to get it working now though thanks. I uninstalled the existing LyZ and updated to the current version. 

After a few teething problems it is now back to usual. 

It’s weird as I've been using those versions together for a couple of years and then a sudden change in functionality!!

Thanks so much for getting back to me. 

Richard
